usaspending_api/accounts/views/federal_accounts_v2.py

```diff
@@ -26,24 +26,29 @@ def get(self, request, pk, format=None):
         # get tas related to FA
         tas_queryset = TreasuryAppropriationAccount.objects.filter(federal_account=fa)
 
-        # get fin based on tas
-        for tas in tas_queryset:
-            financial_account_queryset = \
-                FinancialAccountsByProgramActivityObjectClass.objects.filter(treasury_account=tas)
-
-            # get oc out of financial account
-            for financial_account in financial_account_queryset:
-                oc_list.append(financial_account.object_class)
-
-        # remove non-uniques
-        oc_set = set(oc_list)
-
-        # structure response
-        major_classes = set(oc.major_object_class_name for oc in oc_set)
+        # get fin based on tas, select oc, make distict values
+        # add conditions since '__in' will not work with non-querysets
+        financial_account_queryset = \
```

Comment:
Here, to potentially be able to use `__in`, you could try updating `tas_queryset` to look something like this: `tas_queryset = TreasuryAppropriationAccount.objects.filter(federal_account=fa).values_list('treasury_account_identifier', flat=True).distinct()` (I don't think the `.distinct()` is necessary though).

This will return a list of IDs for the matching entries in `TreasuryAppropriationAccount` and that should work with `__in` regardless of how many come back.